### PR TITLE
feat(docs): update FAQ

### PR DESCRIPTION
**Pull request by achrinza**
This PR is part of a larger initiative to incrementally improve the documentation.

Signed-off-by: Rifa Achrinza <25147899+achrinza@users.noreply.github.com>

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/FAQ.md [hunk 1]
```diff
@@ -47,7 +101,7 @@ reasons:
 - **Improved extensibility** and flexibility. LoopBack 4's core is simpler than
   LoopBack 3.x with well-defined extension points. A lot of responsibility will
   be shifted to extensions (components), which can be JavaScript or TypeScript.
-- Unified tooling. TypeScript developers all use the same IDE: Visual Studio
+- **Unified tooling**. TypeScript developers all use the same IDE: Visual Studio
```
Comment (dhmlau) [inline docs/site/FAQ.md:104]: While you're here, I'm thinking to modify this sentence. We probably notice there are users not using VSCode. :) Maybe we can change it along the line that there are good proportion of TypeScript developers are using VSCode, so we provide some vscode configuration to make developing LoopBack applications more convenient

**Review by dhmlau — COMMENTED** [review]
@achrinza, thanks for the PR. The PR is generally LGTM. I have a comment about VSCode, please see my review comment. Thanks.

File: docs/site/FAQ.md [hunk 2]
```diff
@@ -47,10 +101,9 @@ reasons:
 - **Improved extensibility** and flexibility. LoopBack 4's core is simpler than
   LoopBack 3.x with well-defined extension points. A lot of responsibility will
   be shifted to extensions (components), which can be JavaScript or TypeScript.
-- Unified tooling. TypeScript developers all use the same IDE: Visual Studio
-  Code. The LoopBack ecosystem could someday be filled with useful best
-  practices around that IDE and even great developer plugins. Right now that
-  effort is split between various editors and basically non-existent.
+- **Unified tooling**. A good propportion of TypeScript developers use Visual
```
Comment (dhmlau) [inline docs/site/FAQ.md:104]: typo: 
```suggestion
- **Unified tooling**. A good proportion of TypeScript developers use Visual
```

**Review by dhmlau — APPROVED** [review]
LGTM. Please fix the typo.

File: docs/site/FAQ.md [hunk 3]
```diff
@@ -17,23 +19,75 @@ summary: LoopBack 4 is a completely new framework, also known as LoopBack-Next.
...
+- `@loopback/repository`
+
+  Cross-database ORM and ODM framework. Can be used to create cross-database
+  designs, queries and relations.
+
+- `@loopback/filter`
+
+  Contains LoopBack filter typings and builders. Can be used to build queries on
+  LoopBack API consumers.
+
```
Comment (bajtos) [inline docs/site/FAQ.md:76]: Can we mention recently introduced GraphQL support? See https://github.com/strongloop/loopback-next/tree/master/extensions/graphql

File: docs/site/FAQ.md [hunk 4]
```diff
@@ -64,6 +117,28 @@ projects).
...
+https://github.com/strongloop/loopback-next/issues/1920.
+
+Do you have some other features in mind? Add your feature requests at
+[loopback-next/issues/new](https://github.com/strongloop/loopback-next/issues/new).
+
+## Technical usage
+
 ### Does JavaScript still work?
 
 LoopBack 4 itself is written in [TypeScript](https://www.typescriptlang.org)
```
Comment (bajtos) [inline docs/site/FAQ.md:144]: > LoopBack 4 itself is written in [TypeScript](https://www.typescriptlang.org)
(that compiles to JavaScript), but it supports applications written in both
TypeScript and JavaScript. The documentation assumes you have a basic
understanding of the JavaScript language; and when it says "JavaScript" it is
understood to mean ECMAScript version 6 (ES6).

I am afraid LB4 does not support JavaScript, we require our users to write their code in TypeScript. See #560 and the discussion in #1978 for more information on the challenges involved.

**Review by bajtos — APPROVED** [review]
Great initiative, @achrinza 👏🏻 

The changes looks mostly good, I have few ideas to consider.

**Comment by dhmlau** [comment]
@achrinza, since this PR has 2 approvals, I'm going to merge it. Thanks!

**Pull request merged**